vinyldns/resource_zone.go

```diff
@@ -370,3 +427,77 @@ func zone(d *schema.ResourceData) *vinyldns.Zone {
 
 	return zone
 }
+
+func aclRules(d *schema.ResourceData) []vinyldns.ACLRule {
+	rules := []vinyldns.ACLRule{}
+
+	if r, ok := d.GetOk("acl_rule"); ok {
+		schemaRules := r.(*schema.Set).List()
+
+		for _, rule := range schemaRules {
+			r := rule.(map[string]interface{})
+
+			rules = append(rules, vinyldns.ACLRule{
+				AccessLevel: r["access_level"].(string),
+				Description: r["description"].(string),
+				UserID:      r["user_id"].(string),
+				GroupID:     r["group_id"].(string),
+				RecordMask:  r["record_mask"].(string),
+				RecordTypes: aclRecordTypes(r["record_types"].(*schema.Set)),
```

Comment:
I am not a go guy, but you have `record_types` marked as `Optional` in the Resource, however, it looks like you grab it here `r["record_types"]`.  You do that with the other optional fields so I guess that is ok?  Like does it default to an empty array?

Reply:
@pauljamescleary Yes. As discussed in my previous comment, in this case -- due to the lack of an `omitempty` [here](https://github.com/vinyldns/go-vinyldns/blob/0.9.4/vinyldns/zones_resources.go#L30), `ACLRule.RecordTypes` defaults to `[]`.

I also pushed a commit that includes an acceptance test with an undeclared `record_types` (these are run against a localhost, `docker-compose`ed VinylDNS API).